functions/lib/github.js

```diff
@@ -1,167 +1,202 @@
+        status.state = 'success'
+        status.description = `All good! We have a CLA in file for @${pr.user.login}`
+      } else {
         status.state = 'failure'
-        status.target_url = 'https://sign.the.cla'
-        status.description = 'Cannot evaluate CLA for this PR. ' +
-          'Number of commits exceeds the 250 commit limit. ' +
-          'Please contact contact support@opennetworking.org'
-        return ghClient.repos.createStatus(status)
+        status.description = `We don't have a CLA in file for @${pr.user.login}`
+        status.comment = `Hi @${pr.user.login}, ` +
+          'this is the ONF bot 🤖 I\'m glad you want to contribute to ' +
+          'our projects! However, before accepting your contribution, ' +
+          'we need to ask you to sign a Contributor License Agreement ' +
+          '(CLA). You can do it online, it will take only few minutes:' +
+          '\n\n✒️ 👉 https://cla.opennetworking.org\n\n' +
+          'After signing, make sure to add your Github user ID ' +
+          `\`${pr.user.login}\` to the agreement.`
       }
+    } catch (error) {
+      console.error(error)
+      status.state = 'error'
+      status.description = 'cannot check whitelist'
+    }
 
-      // We need a CLA in file for the PR author (github ID), as well as for all
-      // the identities associated with all commits of this PR.
-      const identities = getPrIdentities(pr, ghClient)
+    // If any error occurred, improve user experience by posting a comment.
+    if (status.state === 'error' && !status.comment) {
+      status.comment =
+        `Unable to verify CLA: ${status.description}. ` +
+        'If the problem persists, please contact support@opennetworking.org ' +
+        ` (\`support-id: ${eventSnapshot.id}\`)`
+    }
 
-      return clalib.checkIdentities(identities).then(result => {
-        if (result.allWhitelisted) {
-          console.log('cla is signed for all commits')
-          status.state = 'success'
-          status.description = 'All good! We have a CLA in file for all contributors in this PR.'
+    // Post status to Github and update event in the DB
+    const statusPromise = octokit.repos
+      .createStatus({
+        owner: pr.base.repo.owner.login,
+        repo: pr.base.repo.name,
+        context: 'onf/cla',
+        sha: pr.head.sha,
+        target_url: 'https://cla.opennetworking.org',
+        description: status.description,
+        state: status.state
+      })
+      .then(() => {
+        status.githubAck = true
+      })
+      .catch(error => {
+        status.githubAck = false
+        status.githubError = JSON.parse(JSON.stringify(error))
+      })
+      .then(() => {
+        event.status = status
+        return eventSnapshot.ref.update(event)
+      })
+      .catch(console.error)
+
+    const contribRef = db.collection('contributions')
+      .doc(event.contributionId)
+    const commentPromise = contribRef.get()
```

Comment:
In a `promise` chain is alway good to have a final `.catch` in case of errors

Reply:
There's already a catch at the end of this chain -- am I missing something?